

#Question 1 : What is Simple Linear Regression (SLR)? Explain its purpose.

Ans - ChatGPT said:

Simple Linear Regression (SLR) is a statistical method used to model the relationship between two variables:

- One independent variable (X) — also called the predictor or explanatory variable.
- One dependent variable (Y) — also called the response or outcome variable.

### Definition
Simple Linear Regression tries to fit a straight line through a set of data points in such a way that the line best represents the relationship between
X and 𝑌.

The general form of the regression equation is:

Y=β0+β1X+ε
###Where:
- 𝑌 = dependent variable (what we want to predict)

- 𝑋 = independent variable (the predictor)

- β0 = intercept (value of Y when X = 0)

- 𝛽1 = slope (change in Y for a one-unit change in X)

- ε = random error term (captures variations not explained by
###Purpose of Simple Linear Regression
1. Prediction
To predict the value of one variable (Y) based on the value of another variable (X).
Example: Predicting a person’s weight based on their height.
2. Understanding Relationships
To quantify and interpret the relationship between two variables.
Example: Determining whether sales increase with advertising expenditure.
3. Trend Estimation
To identify trends or patterns in data over time.
4. Modeling and Decision Making
Helps businesses, researchers, and analysts make informed decisions based on observed relationships.

### Example
Suppose you have data on advertising spending (X) and sales (Y).
Running a simple linear regression might yield the equation:

Sales = 50+5×Advertising

This means:

Even with no advertising (X=0), baseline sales are 50 units.

For every additional dollar spent on advertising, sales increase by 5 units.

---

# Question 2: What are the key assumptions of Simple Linear Regression?

Ans - Key Assumptions of Simple Linear Regression (SLR)

Simple Linear Regression is based on several important assumptions that ensure the validity and reliability of the model’s estimates and predictions. These assumptions relate to the relationship between the independent variable (X) and the dependent variable (Y), as well as the behavior of the residuals (errors).

 1. Linearity

- The relationship between the independent variable (X) and the dependent variable (Y) must be linear.
- This means that a change in X results in a proportional change in Y.
- Graphically: The data points should form a straight-line pattern when plotted.

Example:
If advertising expenditure doubles, sales should approximately double — not increase exponentially or stay constant.

 2. Independence of Errors

- The residuals (errors) should be independent of each other.
- There should be no correlation between consecutive error terms.
- This is especially important in time-series data.

Example:
Errors in one month’s sales prediction should not be related to the next month’s errors.

 3. Homoscedasticity (Constant Variance of Errors)

- The residuals should have constant variance across all levels of X.
- In other words, the spread of residuals should remain roughly the same for all predicted values.

If violated:
You have heteroscedasticity, which can make significance tests unreliable.

Visual check:
A plot of residuals vs. predicted values should show a random scatter, not a funnel or cone shape.

 4. Normality of Errors

- The residuals (errors) should be approximately normally distributed.
- This assumption is important for making valid inferences (e.g., confidence intervals, hypothesis tests).

Check:
Plot a histogram or Q-Q plot of residuals — it should look roughly bell-shaped.

 5. No (or Minimal) Measurement Error in X

- The independent variable (X) is assumed to be measured without error.
- If X has large measurement errors, the regression estimates may be biased and inefficient.

 6. No Significant Outliers

- Outliers can have a strong influence on the regression line, distorting slope and intercept estimates.
- It’s important to identify and handle them appropriately.

---
#Question 3: Write the mathematical equation for a simple linear regression model and explain each term.
Ans - The mathematical equation for a Simple Linear Regression (SLR) model is:
Y=β0+β1X+εY
###Where:
- Y (Dependent variable / Response variable):
This is the variable we want to predict or explain. Examples include sales, exam scores, or house prices.
- X (Independent variable / Predictor variable):
This is the variable used to predict Y. Examples include advertising expenditure, study hours, or house size.
- β0 (Intercept / Constant term):
This represents the expected value of Y when X=0X = 0X=0, i.e., where the regression line crosses the Y-axis.
- β1 (Slope / Regression coefficient):
This indicates the change in Y for a one-unit change in X. It shows the strength and direction of the relationship between X and Y.
- ε (Error term / Residual):
This captures the difference between the observed value of Y and the value predicted by the model. It accounts for all factors affecting Y that are not included in X.
###Estimated Regression Line:
When we use sample data to fit the model, the estimated regression line is written as:

Y^=b0​+b1​X
Where:
- Y = predicted value of Y
- b0​ = estimated intercept
- b1​ = estimated slope

Example:
If the regression equation is:

Y=50+5X

- b0​=50: Even with zero advertising, expected sales are 50 units.

- b1​=5: Each additional unit of advertising increases sales by 5 units.

---

#Question 4: Provide a real-world example where simple linear regression can be applied.
Ans - Scenario:
A real estate company wants to estimate the selling price of houses in a city based on their size (in square feet).

Variables:

Independent Variable (X): Size of the house (in square feet)

Dependent Variable (Y): Selling price of the house (in dollars)

How SLR is applied:

Using historical data on house sizes and their selling prices, the company can fit a simple linear regression model:

Price=𝛽0+𝛽1×Size+𝜀

- The intercept (𝛽0) represents the baseline price of a house (even if its size were zero).

- The slope (β1) represents the expected increase in house price for each additional square foot of size.

- The error term (ε) captures other factors affecting price, such as location, age of the house, and amenities.

Interpretation:
If the fitted model is:

Price = 50,000+200×Size

50,000 = baseline price

200 = every additional square foot adds $200 to the house price

Why this works:
This is a classic example of SLR because there is one predictor (size) and a linear relationship is expected between house size and price. The model can help the company predict prices for new houses, make pricing decisions, or analyze trends.


---


#Question 5: What is the method of least squares in linear regression?
Ans - The method of least squares in linear regression is a statistical technique used to find the best-fit line for a set of data points by minimizing the sum of the squared differences (residuals) between the observed and predicted values. This is done by minimizing the sum of the squares of the vertical distances of the data points from the line. The goal is to find a line that represents the data with the least amount of error, resulting in an equation that minimizes these squared errors.

###How it works
1. Plot the data: The independent variable is plotted on the x-axis and the dependent variable on the y-axis, creating a scatter plot.
2. Find the line of best fit: The method finds the line that best represents the overall trend of the points.
3. Calculate residuals: For each data point, the residual is the vertical distance between the actual y-value and the y-value predicted by the line.
4. Square the residuals: Each residual is squared. This ensures that all errors are positive and that larger errors are penalized more heavily.
5. Sum the squared residuals: The squared residuals are added together to get a single value representing the total squared error for the line.
6. Minimize the sum: The method then finds the specific line (defined by its slope and y-intercept) that makes this sum of squared residuals as small as possible.

What is the resulting equation?

- The final result is a linear equation, often written as \(y=mx+b\), where \(m\) is the slope and \(b\) is the y-intercept of the best-fit line.
- This equation can then be used to make predictions about the dependent variable (\(y\)) for new values of the independent variable (x).

---

Question 6: What is Logistic Regression? How does it differ from Linear Regression?

Ans - Logistic regression is a statistical method for predicting the probability of a categorical outcome, while linear regression is used for predicting a continuous numerical outcome. The main differences lie in their use cases (classification vs. regression), output type (categorical vs. continuous), and the underlying mathematical model (sigmoid curve vs. straight line).

###Logistic Regression

- What it is: A classification algorithm used to predict the probability of a binary outcome (e.g., yes/no, true/false, spam/not spam).
-- How it works: It uses a logistic (sigmoid) function to map the relationship between input variables and a continuous output to a probability between 0 and 1.
- Output: A probability value that is then used to classify the outcome based on a threshold (e.g., above 0.5 is "yes," below 0.5 is "no").
- Example: Predicting whether a customer will click an ad based on their demographics.
###Linear Regression
- What it is: A regression algorithm used to predict a continuous output variable.
- How it works: It models the relationship between independent variables and a dependent variable as a straight line.
- Output: A continuous numerical value, such as a price, temperature, or score.
- Example: Predicting the price of a house based on its size, location, and features.




---

#Question 7: Name and briefly describe three common evaluation metrics for regression models.
Ans - Mean Absolute Error (MAE)

- Description: Measures the average of the absolute differences between the predicted and actual values.

Formula:

𝑀𝐴𝐸=1/𝑛 ∑𝑖=1 𝑛 ∣ 𝑦𝑖−𝑦^𝑖 ∣

- Interpretation: Smaller MAE means predictions are closer to actual values. It treats all errors equally.

2. Mean Squared Error (MSE)

- Description: Measures the average of the squared differences between predicted and actual values.

Formula:
𝑀𝑆𝐸 = 1𝑛 ∑𝑖=1𝑛 (𝑦𝑖−𝑦^𝑖) 2
- Interpretation: Penalizes larger errors more than MAE because errors are squared, making it sensitive to outliers.

3. R-squared (Coefficient of Determination)

- Description: Indicates how well the independent variable(s) explain the variation in the dependent variable.

Formula:

𝑅2=1−(SSres/SStot)
where SSres is the sum of squared residuals and SStot is the total sum of squares.

- Interpretation:

R2=1: Perfect prediction.

R2=0: Model explains none of the variability.

---

#Question 8: What is the purpose of the R-squared metric in regression analysis?

Ans - The purpose of the R-squared (R2R^2R2) metric in regression analysis is to measure how well the independent variable(s) explain the variability in the dependent variable.


- It tells us the proportion of the total variation in the dependent variable that is captured by the regression model.


- Value range: 0≤R2≤1


R2=1: The model explains 100% of the variation—perfect fit.


R2=0: The model explains none of the variation—no predictive power.

Example:
If R2=0.75R = 0.75 when predicting house prices from house size, it means 75% of the variation in house prices can be explained by house size, while 25% is due to other factors not included in the model.

Key point: R-squared is a good indicator of model fit, but it doesn’t tell you whether the model is appropriate or if predictions are unbiased—it just measures explanatory power.
---

#Question 9: Write Python code to fit a simple linear regression model using scikit-learn and print the slope and intercept. (Include your Python code and output in the code box below.)

Ans -

In [1]:
# Import required libraries
from sklearn.linear_model import LinearRegression
import numpy as np

# Example dataset
# X = independent variable (e.g., house size in sq ft)
# y = dependent variable (e.g., house price in $)
X = np.array([500, 700, 1000, 1200, 1500]).reshape(-1, 1)
y = np.array([150000, 200000, 250000, 300000, 350000])

# Create the Linear Regression model
model = LinearRegression()

# Fit the model
model.fit(X, y)

# Print the slope (coefficient) and intercept
print("Slope (Coefficient):", model.coef_[0])
print("Intercept:", model.intercept_)


Slope (Coefficient): 199.0445859872611
Intercept: 54936.30573248412


#Question 10: How do you interpret the coefficients in a simple linear regression model?

Ans - In a simple linear regression model, which has the form:


y=a+bX

where:

y = dependent variable

X = independent variable

a = intercept

b = slope (coefficient)

the coefficients are interpreted as follows:

1. Intercept (a)

- This is the predicted value of y when X=0.

- It represents the baseline or starting point of the regression line.

Example: If predicting house prices based on size, an intercept of 54,936 means that theoretically, a house with 0 sq ft would have a price of $54,936 (though this is just a mathematical baseline, not realistic).

2. Slope / Coefficient (b)

- This represents the change in y for a one-unit increase in X.

- It shows the direction and strength of the relationship between X and Y:

Positive b → as X increases, Y increases

Negative b → as X increases, Y decreases

- Example: A slope of 199.04 means that for each additional 1 sq ft of house size, the predicted price increases by $199.